### ISSUE TITLE
refresh chart labels when using API/Ajax [QUESTION]

### ISSUE THREAD

**Issue by mohcinsarrar** [ConsoleTVs/Charts#16]
i am using consoletvs/charts:6.* with laravel 9
i want to refresh labels and data when calling ajax request

**Comment by mnluribef** [comment]
Hi friend, did you find a solution?

**Comment by sheko016** [comment]
Hello, good afternoon, how are you?

I have the same problem.
I am using laravel chart by API/Ajax. But I have a problem.
When loading for the first time, you see he shows me every month in the lebels.
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/9d184b89-4450-4962-b7aa-cce2f29c735f)

but after updating the graph.
I only want to show in the lebels the months February March April and May
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/3175c44a-451c-4050-8d57-44479bc22440)

I don't want to show every month it doesn't work for me that way.
Is there a way to update the lebels.??

**Comment by sheko016** [comment]
Studying the code a bit I could see that currently Laravel Chart does not update the Labels by API
The file where I saw it was in
https://github.com/ConsoleTVs/Charts/blob/main/src/Views/chartjs/script.blade.php

I add an image.
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/0f14b4ab-19bd-4d96-b1a5-e2deb94ff106)


I can request to be able to collaborate in the project to see if I can make it possible to update the labels.
Greetings thank you

**Comment by mnluribef** [comment]
I came across this post: [https://www.georgebuckingham.com/laravel-livewire-chart-js-realtime/](url) it uses ConsoleTVs/Charts as a base and some helper components to reload the graph with AJAX, i tried it and found out it gives me this error:

Attempt to read property "id" on null, file wan-speed-tests.blade.php, line 5 <div wire:ignore wire:key={{ $chart->id }}>

Any ideas on how to solve this?

**Comment by sheko016** [comment]
Hi guys, I know how to fix it.
I already know how to update the labels.
I would like permission to upload the applied update. In the same way I leave here a print of the edited code.

Let's go first to the file:
https://github.com/ConsoleTVs/Charts/blob/main/src/Classes/BaseChart.php
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/55a397ee-57d7-441a-ab2f-f8553c68ed8e)

and the code is like this.
Annex Print original code

![image](https://github.com/ConsoleTVs/Charts/assets/17069879/b8c37ab3-ea14-4c34-9154-bbb00ff434dc)

an if (count($this->labels) >= 1) {} is performed
to ask if there are new labels if the old ones are not left.



**Now in the file.**
https://github.com/ConsoleTVs/Charts/blob/main/src/Views/chartjs/script.blade.php

the code is like this.
Annex Print original code
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/cf3a1c2c-47b1-46b8-ba88-7691a975ef82)

Now the edited code looks like this.
Annex Print edited code
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/aa0ccbff-d05f-4fbc-951e-bd27115a4ca7)

Now in line 9 the variable data is equal to
`data = data.datasets;
`
and in the lines 36 37 38 39
update as the info comes
```
{{ $chart->id }}.data.datasets = data.datasets;
 if (data.labels) {
     {{ $chart->id }}.data.labels = data.labels;
 }
```


Now I show you how it looks.
as I had it before you can see the screen prints in this comment

https://github.com/ConsoleTVs/Charts/issues/16#issuecomment-1664449354


And the new way is now.

Initial charge
![image](https://github.com/ConsoleTVs/Charts/assets/17069879/5ce2f259-0daf-4e1c-bfc4-80bfc02e6048)

updating via ajax

![image](https://github.com/ConsoleTVs/Charts/assets/17069879/e8419b81-b90e-4235-9d83-6fbcb5233d37)

It only brings me the labels that I want to show.
I only wish that the account administrator could upload those changes, that would be very good.
